lib/block_view_message_test.go

```diff
@@ -2392,13 +2558,410 @@ func TestGroupMessages(t *testing.T) {
+		encryptedMessage = encrypt(testMessage, entry.MessagingPublicKey[:])
+		// Create the corresponding message entry and connect it.
+		unmuteMessageEntry := MessageEntry{
+			m0PublicKey,
+			senderPublicKey,
+			encryptedMessage,
+			tstampNanos,
+			false,
+			MessagesVersion3,
+			m0PublicKey,
+			BaseGroupKeyName(),
+			entry.MessagingPublicKey,
+			NewGroupKeyName(gangKey),
+			nil,
+		}
+		_helpConnectPrivateMessageWithParty(testMeta, m0Priv, unmuteMessageEntry, nil)
+		// m0 is now unmuted and hence:
+		// The message should be successfully added, so we now have:
+		// m0 -> group(sender, recipient, m0, m2)
+		// 	sender: 7
+		//	recipient: 8
+		//  m1: 4
+		// 	m0: 5
+		// 	m2: 2
+		// Since we're passed the ExtraData migration, the entry will have the extra data field. We add it after
+		// transaction is processed as an extra sanity-check.
+		unmuteMessageEntry.ExtraData = make(map[string][]byte)
+		unmuteMessageEntry.ExtraData[MessagesVersionString] = UintToBuf(MessagesVersion3)
+		unmuteMessageEntry.ExtraData[SenderMessagingPublicKey] = m0PublicKey.ToBytes()
+		unmuteMessageEntry.ExtraData[SenderMessagingGroupKeyName] = BaseGroupKeyName().ToBytes()
+		unmuteMessageEntry.ExtraData[RecipientMessagingPublicKey] = entry.MessagingPublicKey.ToBytes()
+		unmuteMessageEntry.ExtraData[RecipientMessagingGroupKeyName] = NewGroupKeyName(gangKey).ToBytes()
+		require.Equal(true, _verifyMessageParty(testMeta, expectedMessageEntries, unmuteMessageEntry, false))
+
+		// Verify the messages AGAIN.
+		_verifyMessages(testMeta, expectedMessageEntries)
+		// Just to sanity-check, verify that the number of messages is as intended.
+		utxoView, err = NewUtxoView(db, params, nil, chain.snapshot)
+		require.NoError(err)
+		messages, _, err = utxoView.GetMessagesForUser(senderPkBytes, chain.blockTip().Height+1)
+		require.NoError(err)
+		require.Equal(7, len(messages))
+		messages, _, err = utxoView.GetMessagesForUser(recipientPkBytes, chain.blockTip().Height+1)
+		require.NoError(err)
+		require.Equal(8, len(messages))
+		messages, _, err = utxoView.GetMessagesForUser(m1PubKey, chain.blockTip().Height+1)
+		require.NoError(err)
+		require.Equal(4, len(messages))
+		messages, _, err = utxoView.GetMessagesForUser(m0PubKey, chain.blockTip().Height+1)
+		require.NoError(err)
+		require.Equal(5, len(messages))
+		messages, _, err = utxoView.GetMessagesForUser(m2PubKey, chain.blockTip().Height+1)
+		require.NoError(err)
+		require.Equal(2, len(messages))
+
+		{
+			// Let us now try to unmute m0 AGAIN
+			// This should produce an error since m0 is already unmuted
+			var unmuteList []*MessagingGroupMember
+			unmuteList = append(unmuteList, &MessagingGroupMember{
+				m0PublicKey,
+				BaseGroupKeyName(),
+				encrypt(privBytes, m0PubKey),
+			})
+			//require.Equal(false, _verifyMessagingKey(testMeta, senderPublicKey, entry))
+			extraData = make(map[string][]byte)
+			extraData[MessagingGroupOperationType] = []byte{byte(MessagingGroupOperationUnmuteMembers)}
```

Comment:
Note that you can define a map in one command if you'd like:
```
		extraData = map[string][]byte{
			MessagingGroupOperationType: []byte(byte(MessagingGroupOperationUnmuteMembers)),
		}
```

Reply:
I know but it's 3 lines vs 2 lines.